src/rust/engine/process_execution/src/fork_exec.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::io;
+use std::sync::Arc;
+use std::time::Duration;
+
+use tokio::sync::RwLock;
+
+use crate::ManagedChild;
+
+/// Spawn a subprocess safely given that binaries may be written by this process.
+pub async fn spawn_process(
```

Comment:
From pairing review: Move this refactor to separate PR.

Reply:
@benjyw: I refactored the `spawn_process` helper in a separate PR https://github.com/pantsbuild/pants/pull/20944as discussed.